shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -152,23 +152,24 @@ public void createAndroidViewForPlatformView(
             throw new IllegalStateException(
                 "The Android view returned from PlatformView#getView() was already added to a parent view.");
           }
-          platformViews.put(request.viewId, view);
+          platformViews.put(request.viewId, platformView);
         }
 
         @Override
         public void disposeAndroidViewForPlatformView(int viewId) {
           // Hybrid view.
-          final View platformView = platformViews.get(viewId);
+          final PlatformView platformView = platformViews.get(viewId);
           final FlutterMutatorView parentView = platformViewParent.get(viewId);
           if (platformView != null) {
             if (parentView != null) {
-              parentView.removeView(platformView);
+              parentView.removeView(platformView.getView());
             }
             platformViews.remove(viewId);
+            platformView.dispose();
           }
 
           if (parentView != null) {
-            ((FlutterView) flutterView).removeView(parentView);
```

Comment:
this should still work, no?

Reply:
`flutterView` may be null at this point, but calling `getParent` will always work